backend/BusinessLogic/Services/VideoService/VideoService.cs

```diff
@@ -99,7 +99,7 @@ public async Task<Video> CompleteUpload(Guid userId, string fileName)
 
         public async Task DeleteVideo(Video video, Guid userId)
         {
-            string snapshotPath = Path.Combine(Path.Combine(_uploadPath, userId.ToString()), "Snapshots\\" + video.Id + ".png");
+            string snapshotPath = Path.Combine(Path.Combine(_uploadPath, userId.ToString()), Path.Combine("Snapshots", video.Id + ".png"));
 
             if (!File.Exists(video.Path))
```

Comment:
General comment: 

https://user-images.githubusercontent.com/31959982/119030190-3c2f9280-b9b2-11eb-831b-efddaa30271d.mp4

Some flickering action going on

Reply:
You should wait for a response before rendering anything